### PR TITLE
add USPS dataset

### PR DESCRIPTION
**Pull request by youkaichao**
I have implemented the USPS dataset as requested in #298 .

### CODE REVIEW

**Comment by youkaichao** [comment]
anyone working on this thread?

File: torchvision/datasets/usps.py [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+
+        if download and not os.path.exists(full_path):
+            self.download()
+
+        import bz2
+        fp = bz2.open(full_path)
+
+        datas = []
+        labels = []
+        for line in tqdm.tqdm(
```
Comment (varunagrawal) [inline torchvision/datasets/usps.py]: Is tqdm part of torchvision?

File: torchvision/datasets/usps.py [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+            pixels = [float(x.split(':')[-1]) for x in pixels]
+            im = np.asarray(pixels).reshape((16, 16))
+            im = (im + 1) / 2 * 255
+            im = im.astype(dtype=np.uint8)
+            datas.append(im)
+            labels.append(int(label) - 1)
+
+        assert len(
+            labels
+        ) == self.total_images, 'total number of images are wrong! maybe the download is corrupted?'
```
Comment (varunagrawal) [inline torchvision/datasets/usps.py]: Please split this assert as 
```python
assert len(labels) == self.total_images,
    'total number of images are wrong! maybe the download is corrupted?'
```

File: torchvision/datasets/usps.py [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+                fp, desc='processing data', total=self.total_images):
+            label, *pixels = line.decode().split()
+            pixels = [float(x.split(':')[-1]) for x in pixels]
+            im = np.asarray(pixels).reshape((16, 16))
+            im = (im + 1) / 2 * 255
+            im = im.astype(dtype=np.uint8)
+            datas.append(im)
+            labels.append(int(label) - 1)
+
+        assert len(
```
Comment (varunagrawal) [inline torchvision/datasets/usps.py]: This assert does not make sense. The dataset only has 10 labels as per the [webpage](https://www.csie.ntu.edu.tw/~cjlin/libsvmtools/datasets/multiclass.html#usps) and over 7000 total images.

Semantically it seems wrong.

File: torchvision/datasets/usps.py (hunk 2 above)
Comment (varunagrawal) [reply torchvision/datasets/usps.py]: Use a formatter such as autopep8 to get this consistent with PEP8

File: torchvision/datasets/usps.py (hunk 1 above)
Comment (youkaichao) [reply torchvision/datasets/usps.py]: I will switch to ``from torch.utils.model_zoo import tqdm``.

File: torchvision/datasets/usps.py (hunk 2 above)
Comment (youkaichao) [reply torchvision/datasets/usps.py]: Okay. This is the formating result of flake8 and I also think it is ugly.

File: torchvision/datasets/usps.py (hunk 3 above)
Comment (youkaichao) [reply torchvision/datasets/usps.py]: Well, what I mean by `labels` is not "how many labels" but "labels of images". This is confusing, I rename it as `targets` to be more clear.

**Comment by youkaichao** [comment]
Is this thread active?

**Comment by varunagrawal** [comment]
@youkaichao please be patient. We're getting around to reviewing all the PRs.

**Comment by youkaichao** [comment]
well, almost 1 month now...

**Comment by youkaichao** [comment]
Anyone working in this thread? @varunagrawal

**Comment by youkaichao** [comment]
Anyone working in this thread?

**Comment by fmassa** [comment]
@youkaichao I'm reviewing it today, sorry for the delay!

**Comment by fmassa** [comment]
@youkaichao I've fixed a few minor things with the PR in https://github.com/pytorch/vision/pull/961, thanks a lot for the PR and sorry for the delay!
